### ISSUE TITLE
ghc cannot compile executables on Ubuntu yakkety amd64

### ISSUE THREAD

**Issue by andersk** [hvr/multi-ghc-travis#45]
Because yakkety’s GCC now [builds with -fPIE by default](https://lists.ubuntu.com/archives/ubuntu-devel/2016-April/039331.html), GHC in ppa:hvr/ghc is no longer able to compile executables:

```console
$ echo 'main = putStrLn "Hello, world!"' > hello.hs
$ /opt/ghc/7.10.3/bin/ghc hello.hs 
[1 of 1] Compiling Main             ( hello.hs, hello.o )
Linking hello ...
/usr/bin/ld.bfd.real: hello.o: relocation R_X86_64_32S against `stg_bh_upd_frame_info' can not be used when making a shared object; recompile with -fPIC
hello.o: error adding symbols: Bad value
collect2: error: ld returned 1 exit status
```

To work around this error, Ubuntu applied a [patch disabling PIE on amd64 and ppc64el](http://launchpadlibrarian.net/255267967/ghc_7.10.3-7_7.10.3-7ubuntu1.diff.gz) to yakkety’s GHC. Consider patching the PPA packages similarly.

**Comment by andersk** [comment]
Additionally, Cabal needs haskell/cabal#3443.

**Comment by hvr** [comment]
@andersk fyi, since yakkety went into release freeze recently, I've started uploading Yakkety packages

**Comment by andersk** [comment]
Cool. It looks like your yakkety packages are patched to work on yakkety (I checked 7.10.3 and 8.0.1), so this can be closed.

**Comment by hvr** [comment]
Indeed, I used the same method as Ubuntu. However, just as with Ubuntu's `ghc`, `ghc -shared` is broken; this will most likely need to be fixed in GHC upstream.
